### ISSUE TITLE
GCS config printout truncated in the log

### ISSUE THREAD

**Issue by philip-galera** [codership/galera#124]
When messing with wsrep_provider_options, for example

```
set global wsrep_provider_options='base_port=13400';
```

, I got the following line in the mysqld error log:

```
140909 15:16:15 [Note] WSREP: Passing config to GCS: base_host = 127.0.0.1; base_port = 3400; cert.log_conflicts = no; debug = no; evs.inactive_check_period = PT0.5S; evs.inactive_timeout = PT15S; evs.join_retrans_period = PT1S; evs.max_install_timeouts = 1; evs.send_window = 4; evs.stats_report_period = PT1M; evs.suspect_timeout = PT5S; evs.user_send_window = 2; evs.view_forget_timeout = PT24H; gcache.dir = /tmp/vardir1/master-data/; gcache.keep_pages_size = 0; gcache.mem_size = 0; gcache.name = /tmp/vardir1/master-data//galera.cache; gcache.page_size = 128M; gcache.size = 128M; gcs.fc_debug = 0; gcs.fc_factor = 1.0; gcs.fc_limit = 16; gcs.fc_master_slave = no; gcs.max_packet_size = 64500; gcs.max_throttle = 0.25; gcs.recv_q_hard_limit = 9223372036854775807; gcs.recv_q_soft_limit = 0.25; gcs.sync_donor = no; gmcast.segment = 0; gmcast.version = 0; pc.announce_timeout = PT3S; pc.checksum = false; pc.ignore_quorum = false; pc.ignore_sb = false; pc.npvo = false; pc.version = 0; pc.wait_prim = true; pc.wait_prim_timeout = P30S; pc.we
```

The line seems truncated.

**Comment by roberthunt** [comment]
Could this be a bug/limitation in MySQL related to `SHOW VARIABLES` having a maximum length for the value returned or something? I was trying to confirm that the Galera config between nodes was identical and noticed the output was truncated when running `SHOW VARIABLES LIKE 'wsrep_provider_options';`

**Comment by scolebrook** [comment]
@roberthunt I've noticed this too. It appears to be limited to 1024 bytes after we upgraded to MySQL 5.7. During the upgrade, when some nodes were on 5.6.34 and one on 5.7.16. The 5.6 nodes showed the full wsrep_provider_options value while the 5.7 node was truncated at 1024 bytes. So it seems to be a change in 5.7 related to system variable length which I can't find documented or even commented on anywhere. I've been looking for a new configuration option to adjust this somewhere but can't find anything. Every system variable with a value of 1024 doesn't have anything to do with this.

Some of the options that are truncated are the paths to the ssl certificates for encryption of replication traffic. It appears to still be encrypted even though the value of the variable is truncated. So I think this is just a problem with storing the options used during start up in the variable. I don't think it actually impacts operations, unless you need to get the a value inside wsrep_provider_options for any reason.

**Comment by roberthunt** [comment]
Yes, that was my conclusion - it's just the truncation of the reported value rather than the config being wrong as it cuts off half way through some options or values. This MySQL bug seems to be similar: https://bugs.mysql.com/bug.php?id=76680
